pkg/skaffold/runner/deployer.go

```diff
@@ -271,3 +290,22 @@ func validateKubectlFlags(flags *latest.KubectlFlags, additional latest.KubectlF
 	}
 	return nil
 }
+
+/* The Cloud Run deployer for apply. Used when Cloud Run is specified. */
+func getCloudRunDeployer(runCtx *runcontext.RunContext, labeller *label.DefaultLabeller) (*cloudrun.Deployer, error) {
+	var region string
+	var defaultProject string
+	for _, d := range runCtx.DeployConfigs() {
+		if d.CloudRunDeploy != nil {
+			crDeploy := d.CloudRunDeploy
+			if region != "" && region != crDeploy.Region {
+				return nil, fmt.Errorf("expected all Cloud Run deploys to be in the same region, found deploys to %s and %s", region, crDeploy.Region)
```

Comment:
nit: These validations are likely better suited to be in https://github.com/GoogleContainerTools/skaffold/blob/main/pkg/skaffold/schema/validation/validation.go

as these schema checks occur earlier in skaffold

Reply:
Makes sense thanks for clarifying, fine to leave as is for the reasons stated.